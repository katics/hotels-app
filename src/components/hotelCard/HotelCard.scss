.hotel-card {
  background-color: rgb(241, 241, 241);
}
.pointer {
  cursor: pointer;
}
