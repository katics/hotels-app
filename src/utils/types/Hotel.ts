export interface IHotel {
  name?: string;
  // adress?: string;
  // stars?: number;
}
