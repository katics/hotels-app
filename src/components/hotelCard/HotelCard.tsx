import { FC } from "react";
import { NavLink } from "react-router-dom";
import { APP_ROUTES } from "../../utils/AppRoutes";

import { Hotel } from "../../utils/types/Hotel";
import "./HotelCard.scss";

interface HotelData {
  hotel: Hotel;
}

const HotelCard: FC<HotelData> = props => {
  return (
    <div className="col-md-3 py-2">
      <div className=" hotel-card card card-body h-100">
        <img
          className="card-img-top"
          src={props.hotel.image}
          alt="Card image"
        />
        <h4 className="card-title h-100">{props.hotel.name}</h4>
        <div>
          <label>Country:</label>
          <span> {props.hotel.country}</span>
        </div>
        <div>
          <label>City:</label>
          <span> {props.hotel.city}</span>
        </div>
        <div>
          <label>Stars:</label>
          <span>{props.hotel.stars}</span>
        </div>
        <NavLink to={`${APP_ROUTES.dashboard}/${props.hotel.id}`}>
          Details
        </NavLink>
      </div>
    </div>
  );
};

export default HotelCard;
