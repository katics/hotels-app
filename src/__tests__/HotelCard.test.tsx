import { render } from "@testing-library/react";
import { StaticRouter } from "react-router-dom";

import HotelCard from "../components/hotelCard/HotelCard";

import { singleHotel } from "../utils/mocks/Hotel.mock";

describe("Hootel Card", () => {
  it("Should match snapshot", () => {
    const { container } = render(
      <StaticRouter>
        <HotelCard
          userId={1}
          hotel={singleHotel}
          addRemoveFavHotel={() => {
            return true;
          }}
        />
      </StaticRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Should show hotel image", async () => {
    const hotelCard = render(
      <StaticRouter>
        <HotelCard
          userId={1}
          hotel={singleHotel}
          addRemoveFavHotel={() => {
            return true;
          }}
        />
      </StaticRouter>
    );
    const hotelCardImage: any = await hotelCard.findByTestId(
      "HotelCardImgTestId"
    );
    expect(hotelCardImage.src).toContain(singleHotel.image);
  });
  it("Should display hotel city", async () => {
    const hotelCard = render(
      <StaticRouter>
        <HotelCard
          userId={1}
          hotel={singleHotel}
          addRemoveFavHotel={() => {
            return true;
          }}
        />
      </StaticRouter>
    );
    const hotelCity = await hotelCard.findByText(singleHotel.city);
    expect(hotelCity.innerHTML).toContain(singleHotel.city);
  });
});
