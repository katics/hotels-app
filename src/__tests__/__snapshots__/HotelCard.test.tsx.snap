// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hootel Card Should match snapshot 1`] = `
<div
  class="col-md-3 py-2"
>
  <div
    class=" hotel-card card card-body h-100"
  >
    <img
      alt="Card image"
      class="card-img-top"
      data-testid="HotelCardImgTestId"
      src="http://127.0.0.1:8000/media/images/54129602.jpg"
    />
    <h4
      class="card-title h-100"
    >
      Courtyard by Marriott Belgrade City Center
    </h4>
    <div>
      <label>
        Country:
      </label>
      <span>
         
        Serbia
      </span>
    </div>
    <div>
      <label>
        City:
      </label>
      <span>
         
        Belgrade
      </span>
    </div>
    <div>
      <label>
        Stars:
      </label>
      <span>
         
        4
      </span>
    </div>
    <div>
      <a
        href="/dashboard/42"
      >
        Details
      </a>
       
      <div
        class="float-right pointer"
      >
        <svg
          fill="royalblue"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 2.748l-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
